server/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java

```diff
@@ -54,7 +54,14 @@ public static Environment prepareEnvironment(
         loadOverrides(output, properties);
         output.put(input);
         replaceForcedSettings(output);
-        output.replacePropertyPlaceholders();
+        try {
+            output.replacePropertyPlaceholders();
+        } catch (Exception e) {
+            throw new SettingsException(
+                "Failed to replace property place holders from '" + configFile.getFileName() + "' " + "caused by: " + e.getMessage(),
```

Comment:
Nits:
- No need for the inner exception message, we're including the whole inner exception here.
- "Placeholders" is one word
- We tend to use `[]` to quote things in messages like this

```suggestion
                "Failed to replace property placeholders from [" + configFile.getFileName() + "]",
```